libenv/unix_iface.c

```diff
@@ -677,6 +676,57 @@ static void FindV6InterfacesInfo(EvalContext *ctx, Rlist **interfaces, Rlist **h
             }
         }
 
+
+        const char *const stripped_ifconfig_line =
+            TrimWhitespace(ifconfig_line);
+        if (StringStartsWith(stripped_ifconfig_line, "ether"))
+        {
+            Seq *const ether_line =
+                SeqStringFromString(stripped_ifconfig_line, ' ');
+            const size_t length = SeqLength(ether_line);
+
+            if (length < 2
+                || !StringSafeEqual_IgnoreCase(SeqAt(ether_line, 0), "ether"))
+            {
+                Log(LOG_LEVEL_ERR,
+                    "Failed to parse hw_mac address for: %s",
+                    current_interface);
+            }
+            else
+            {
+                const char *const hw_mac = SeqAt(ether_line, 1);
+
+                if (!RlistContainsString(*hardware, hw_mac))
+                {
+                    Log(LOG_LEVEL_VERBOSE,
+                        "Adding MAC address: %s for %s",
+                        hw_mac,
+                        current_interface);
+
+                    RlistAppend(hardware, hw_mac, RVAL_TYPE_SCALAR);
+                    assert(RlistContainsString(*hardware, hw_mac));
```

Comment:
Please move this assert into the `RlistAppend()` function.

Reply:
If so, then should there be a `RlistAppendString` function ?

Reply:
Sounds like a plan to me!

Reply:
But that sounds like something that's a bit out of scope of this PR>